### DISCUSSION TITLE
ETF Dividend Reinvestment Plans (DRIP)

### DISCUSSION THREAD

**Discussion by jaysbeekay** [ghostfolio/ghostfolio#3358] [Q&A]
Hello,

Firstly thanks for this great piece of software - I am in the process of using it to track all my assets and had a question around the best way to handle automatic dividend reinvestment plans?

Is the best approach to:

1. import all dividends using the native functionality and then create a correlating BUY activity with a unit price of 0 but capturing the correct quantity received
2. Create a BUY activity with a unit price of 0 for each dividend received but capturing the correct quanity received
3. Create a DIVIDEND activity with unit price of 0 and but capturing the correct quantity received

Thanks!

**Comment by cjboyle** [comment]
Best approach is no.1, but set the BUY price to the value per share as received. In other words, you didn't receive cash AND shares, you received cash which was then used to purchase more shares.

Your broker should provide the unit cost of the new shares, but if not, ((Dividend / Number Of Shares)) should be close enough.
